chess2.cpp

```diff
@@ -0,0 +1,882 @@
+        if (row != 0 && col != 7 && board[row - 1][col + 1] <= 0) pseudomoves.insert(pseudomoves.begin(),push(row, col, row - 1, col + 1));
+        if (row != 0 && col != 0 && board[row - 1][col - 1] <= 0) pseudomoves.insert(pseudomoves.begin(),push(row, col, row - 1, col - 1));
+    }
+    //returnpseudomoves;
+}
+
+
+int Chess2::evaluation() {
+    const short int pawn = 95,bishop = 330,knight = 320,rook = 500,queen = 900,king = 2000;
+    const int pawnt[8][8] = {
+     {0,  0,  0,  0,  0,  0,  0,  0},
+    {50, 50, 50, 50, 50, 50, 50, 50},
+    {10, 10, 20, 30, 30, 20, 10, 10},
+     {5,  5, 10, 45, 45, 10,  5,  5},
+     {0,  0,  0, 20, 20,  0,  0,  0},
+    {5, -5,-10,  0,  0,-10, -5,  5},
+    {5, 10, 10,-20,-20, 10, 10,  5},
+    {0,  0,  0,  0,  0,  0,  0,  0}
+    };
+
+    const int bishopt[8][8] = {
+        {-20,-10,-10,-10,-10,-10,-10,-20},
+        {-10,  0,  0,  0,  0,  0,  0,-10},
+        {-10,  0,  5, 10, 10,  5,  0,-10},
+        {-10,  5,  5, 10, 10,  5,  5,-10},
+        {-10,  0, 10, 10, 10, 10,  0,-10},
+        {-10, 10, 10, 10, 10, 10, 10,-10},
+        {-10,  5,  0,  0,  0,  0,  5,-10},
+        {-20,-10,-10,-10,-10,-10,-10,-20},
+    };
+
+    const int knightt[8][8] = {
+    {-50,-40,-30,-30,-30,-30,-40,-50},
+    {-40,-20,  0,  0,  0,  0,-20,-40},
+    {-30,  0, 10, 15, 15, 10,  0,-30},
+    {-30,  5, 15, 20, 20, 15,  5,-30},
+    {-30,  0, 15, 20, 20, 15,  0,-30},
+    {-30,  5, 10, 15, 15, 10,  5,-30},
+    {-40,-20,  0,  5,  5,  0,-20,-40},
+    {-50,-40,-30,-30,-30,-30,-40,-50},
+    };
+
+    const int queent[8][8] = {
+        {-20,-10,-10, -5, -5,-10,-10,-20},
+        {-10,  0,  0,  0,  0,  0,  0,-10},
+        {-10,  0,  5,  5,  5,  5,  0,-10},
+         {-5,  0,  5,  5,  5,  5,  0, -5},
+          {0,  0,  5,  5,  5,  5,  0, -5},
+        {-10,  5,  5,  5,  5,  5,  0,-10},
+        {-10,  0,  5,  0,  0,  0,  0,-10},
+        {-20,-10,-10, -5, -5,-10,-10,-20}
+    };
+    const int kingt[8][8] = {
+        {-30,-40,-40,-50,-50,-40,-40,-30},
+        {-30,-40,-40,-50,-50,-40,-40,-30},
+        {-30,-40,-40,-50,-50,-40,-40,-30},
+        {-30,-40,-40,-50,-50,-40,-40,-30},
+        {-20,-30,-30,-40,-40,-30,-30,-20},
+        {-10,-20,-20,-20,-20,-20,-20,-10},
+        {20, 20,  0,  0,  0,  0, 20, 20},
+        {20, 30, 10,  0,  0, 10, 30, 20 },
+    };
+    const int rookt[8][8] = {
+      {0,  0,  0,  0,  0,  0,  0,  0},
+      {5, 10, 10, 10, 10, 10, 10,  5},
+     {-5,  0,  0,  0,  0,  0,  0, -5},
+     {-5,  0,  0,  0,  0,  0,  0, -5},
+     {-5,  0,  0,  0,  0,  0,  0, -5},
+     {-5,  0,  0,  0,  0,  0,  0, -5},
+     {-5,  0,  0,  0,  0,  0,  0, -5},
+      {0,  0,  0,  5,  5,  0,  0,  0}
+    };
+    int score = 0;
+
+    for (int i = 0; i < 8; i++) {
+        for (int j = 0; j < 8; j++) {
+            if (!board[i][j]) continue;
+            if (board[i][j] == 1) {
+                score-=pawnt[i][j];
```

Comment:
these calculations should be part of individual methods. Also classes. For example PawnClass